src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1535,2156 @@ interface MPSMatrixMultiplication {
+	interface MPSMatrixDecompositionLU {
+		[Export ("initWithDevice:rows:columns:")]
+		IntPtr Constructor (IMTLDevice device, nuint rows, nuint columns);
+
+		[Export ("encodeToCommandBuffer:sourceMatrix:resultMatrix:pivotIndices:status:")]
+		void EncodeToCommandBuffer (IMTLCommandBuffer commandBuffer, MPSMatrix sourceMatrix, MPSMatrix resultMatrix, MPSMatrix pivotIndices, [NullAllowed] IMTLBuffer status);
+
+		// inlining ctor from base class
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSMatrixUnaryKernel))]
+	[DisableDefaultCtor] // According to docs needs a Metal Device so initWithDevice: makes more sense.
+	interface MPSMatrixDecompositionCholesky {
+		[Export ("initWithDevice:lower:order:")]
+		IntPtr Constructor (IMTLDevice device, bool lower, nuint order);
+
+		[Export ("encodeToCommandBuffer:sourceMatrix:resultMatrix:status:")]
+		void EncodeToCommandBuffer (IMTLCommandBuffer commandBuffer, MPSMatrix sourceMatrix, MPSMatrix resultMatrix, [NullAllowed] IMTLBuffer status);
+
+		// inlining ctor from base class
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface MPSMatrixCopyDescriptor {
+		[Static]
+		[Export ("descriptorWithSourceMatrix:destinationMatrix:offsets:")]
+		MPSMatrixCopyDescriptor Create (MPSMatrix sourceMatrix, MPSMatrix destinationMatrix, MPSMatrixCopyOffsets offsets);
+
+		[Export ("initWithDevice:count:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, nuint count);
+
+		[Export ("setCopyOperationAtIndex:sourceMatrix:destinationMatrix:offsets:")]
+		void SetCopyOperation (nuint index, MPSMatrix sourceMatrix, MPSMatrix destinationMatrix, MPSMatrixCopyOffsets offsets);
+
+		[Export ("initWithSourceMatrices:destinationMatrices:offsetVector:offset:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (MPSMatrix[] sourceMatrices, MPSMatrix[] destinationMatrices, [NullAllowed] MPSVector offsets, nuint byteOffset);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSKernel))]
+	[DisableDefaultCtor] // There is a DesignatedInitializer, file a bug if needed.
+	interface MPSMatrixCopy {
+		[Export ("initWithDevice:copyRows:copyColumns:sourcesAreTransposed:destinationsAreTransposed:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, nuint copyRows, nuint copyColumns, bool sourcesAreTransposed, bool destinationsAreTransposed);
```

Comment:
@dalexsoto since we moved to `bool AreSourcesTransposed { get; }` and `bool AreDestinationsTransposed { get; }` we'd better update the parameters names for consistency `bool sourcesAreTransposed, bool destinationsAreTransposed` -> `bool areSourcesTransposed, bool areDestinationsTransposed`